src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -90,10 +98,18 @@
    * @return completable future with {@link CompositePurchaseOrder} object with populated uuid on success or an exception if processing fails
    */
   public CompletableFuture<CompositePurchaseOrder> createPurchaseOrder(CompositePurchaseOrder compPO) {
-    return setPoNumberIfMissing(compPO)
-      .thenCompose(v -> poNumberHelper.checkPONumberUnique(compPO.getPoNumber()))
-      .thenCompose(v -> createPOandPOLines(compPO))
-      .thenApply(this::populateOrderSummary);
+    return protectionHelper.isOperationRestricted(compPO.getAcqUnitIds())
+      .thenCompose(isProtected -> {
+        if(isProtected) {
+          throw new HttpException(HttpStatus.HTTP_FORBIDDEN.toInt(), USER_HAS_NOT_PERMISSIONS);
+        } else {
+          return setPoNumberIfMissing(compPO)
+            .thenCompose(v -> poNumberHelper.checkPONumberUnique(compPO.getPoNumber()))
+            .thenCompose(v -> createPOandPOLines(compPO))
+            .thenCompose(v -> createUnitAssignments(compPO))
```

Comment:
The assignments are created after the Order,so if the User doesn't have the permission "orders-storage.acquisitions-unit-assignments.item.post", the order is still created, but without assignments. We should either 
1. clean up the order
2. Check the assignment permission before creating the order

Reply:
it should be 2) per https://wiki.folio.org/display/FOLIJET/Acquisitions+Units#AcquisitionsUnits-RestrictingCreation/ModificationofRecords

Reply:
Assignments creation moved before order creation. So if permission is not presented flow should break. I updated unit test to cover this flow - if POST assignment was made then POST order was made too, otherwise - both requests are absent.

Reply:
This still needs a change because, id will not be present in the incoming POST request, so getId() method used below will not have any value  
https://github.com/folio-org/mod-orders/blob/551e598bbad5494188e911b1344187ee5f440e39/src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java#L340

Also present in the unit test logs:
![Screen Shot 2019-07-25 at 8 37 01 AM](https://user-images.githubusercontent.com/41596468/61874885-75327500-aeb7-11e9-9827-8e99c1555f43.png)

Reply:
@KVupp,

Reply:
To make it possible, we have to remove `recordId` foreign key

Reply:
I think it's better to check the user permission before, instead of removing the foreign key constraint.

Reply:
Another scenario to consider is, what if we create the Order, and then assignments API fails for some reason(even thought it has the required permissions), then an order will be created without any acq units, I think in this scenario we should clean up the created order and return an error

Reply:
> I think it's better to check the user permission before, instead of removing the foreign key constraint.

@KVupp  Look at last comment https://wiki.folio.org/display/FOLIJET/Acquisitions+Units

Reply:
then other thing I can think of is to create a new /validate endpoint to see if the user has assignment permission.

Reply:
OK, I'm not sure there are any great options here... I gave it some thought tonight and wrote up the options on the wiki page for comparison (https://wiki.folio.org/display/FOLIJET/Acquisitions+Units#AcquisitionsUnits-OrderofOperationsDuringRecordCreation)

I've added a 4th option not discussed yet.  Please take a look; we can discuss tomorrow.  If I missed anything feel free to add it to the page or comments section.

Reply:
Updated:  https://wiki.folio.org/display/FOLIJET/Acquisition+Units?focusedCommentId=25727487#AcquisitionUnits-OrderofOperations/Chicken&EggProblem